aws-ecs/README.md

```diff
@@ -0,0 +1,408 @@
+
+# Service Discovery and Load Balancing with Weave on Amazon's ECS #
+
+## What you will build ##
+
+Weave allows you to focus on developing your application, rather than your
+infrastructure.
+
+In this example you will be using Weave for service discovery and load balancing
+between [ECS containers](http://aws.amazon.com/ecs/).
+
+You will use two type of containerized microservices: HTTP Servers and Data
+Producers.
+
+<!---
+Github does not display svg images in READMEs for security reasons
+https://github.com/isaacs/github/issues/316
+
+However, I left it since svg adapts much better to different displays and the
+final user should use Weavework's website anyways.
+-->
+![system diagram](img/diagram.svg)
+
+Unsurprisingly, the HTTP Servers will serve data produced from the Data
+Producers.
+
+This is a very common pattern, whose implementation phase requires answering a
+few questions, among which are:
+
+* Service discovery: How does an HTTP Server find a Data Producer to connect to?
+* Load balancing/fault tolerance: How can the HTTP Server's make uniform use of
+  all the Data Producers?
```

Comment:
@2opremio ok, yes, that's what I meant to answer, may it was a bit too short... Basically with plain ECS, you have to care about declaring what ports your internal service will use and client have to lookup environment variables that have host IPs and remapped port number of those services. With weave you just assume default ports and lookup IPs in DNS.